### PR TITLE
Dynamically loading test blocks

### PR DESCRIPTION
**Pull request by luislpv22**
Some progress made on issue #36 

A bunch of tests are failing now because we are testing all blocks under /data/blocks, and some of them are giving errors

We might check them.

### CODE REVIEW

File: packages/core/src/__tests__/snapshots.test.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      const jsxLite = componentToJsxLite(backToJsxLite);
+      expect(jsxLite).toMatchSnapshot();
+    });
+  });
+});
+
+describe('Html', () => {
+  testCases.forEach((data: TestData) => {
+    test(data.name, () => {
+      const jsx: string = require(data.file);
```
Comment (mandx) [inline packages/core/src/__tests__/snapshots.test.ts:35]: I don't think we need to use `require` and invoke Node's module resolution algorithm, using `fs.readFileSync` should be enough, something like:
```suggestion
      const jsx = fs.readFileSync(data.file, 'utf8');
```

**Review by mandx — CHANGES_REQUESTED** [review]
I added some minor suggestions, in general this is a great starting point. I'll look into why the tests are failing; looks like for some reason no code is being transpiled, so code that expect non-empty strings is crashing.

**Comment by luislpv22** [comment]
I've added your suggestions. Those are the blocks failing in tests:

 - All parsers › builder-render-block

```js
SyntaxError: JSON5: invalid end of input at 1:1

      490 |   );
      491 |   try {
    > 492 |     return JSON5.parse(toParse);
```
 - All parsers › builder-render-content

```js
SyntaxError: JSON5: invalid end of input at 1:1

      490 |   );
      491 |   try {
    > 492 |     return JSON5.parse(toParse);
```
- Html parser › form

```js
SyntaxError: unknown: Unexpected token, expected ";" (1:12)

    > 1 | let _ = get submissionState() {
        |             ^
      2 |   return Builder.isEditing && props.previewState || state.state;
      3 | }
```
- Html parser › columns

```js
SyntaxError: unknown: Unexpected token, expected ";" (3:13)

      1 | let _ = // TODO: These methods are not used right now, but they will be when
      2 | // support for dynamic CSS lands
    > 3 | getColumns() {
        |              ^
```

**Comment by mandx** [comment]
Thanks @luislpv22 I'll take another look later today :)

**Comment by steve8708** [comment]
hey @mandx - any update here?

**Comment by mandx** [comment]
@luislpv22 Looks like this PR has diverged a lot from the main branch; I'll suggest two things to get this restarted:

- Start over with a new/fresh branch, copy code as needed from this branch.
- Try make _additive_ changes, that is, let's introduce new unit-tests modules/files/snapshots instead of overwriting existing ones; worst case, we would be testing the same thing twice, but this will allow avoiding rebase/merge hell since the new PR will be mostly adding code instead of modifying existing code. We can always remove duplicated tests in a secondary PR.

**Comment by steve8708** [comment]
closing out due to inactivity to keepour PR list clean, please feel free to reopen @luislpv22 if you'd like to continue on this

**Pull request closed without merging**